### PR TITLE
05 - Sign up add redux

### PR DESCRIPTION
**Pull request by ValesarroRS**
#### Trello board reference:

- [Trello Card #4] 

(https://trello.com/c/A0PB6RZY)

#### Description:

- Add and set up redux to the framework
- Add api calls to register
- Adapt form to fit the API requirements
- Add tailwindcss-prettier plugin 

#### Reviewers:

- @diebarral 
- @joaquincollado 
- @kevin-rodriguez 
- @prina92 
- @DJA89 

### Screen recording:

https://user-images.githubusercontent.com/88793750/230964051-e82efcd6-b752-425d-8674-332095947ab2.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,19 +3,25 @@
...
     "@types/jest": "^27.5.2",
     "@types/node": "^16.18.12",
     "@types/react": "^18.0.28",
     "@types/react-dom": "^18.0.11",
+    "@types/react-redux": "^7.1.25",
     "classnames": "^2.3.2",
+    "dotenv": "^16.0.3",
     "eslint-plugin-jest-dom": "^4.0.3",
+    "eslint-plugin-tailwindcss": "^3.11.0",
     "eslint-plugin-testing-library": "^5.10.2",
```
Comment (joaquincollado) [inline package.json]: Those are dev dependencies, same with the testing ones

File: src/components/shared/InputFields/index.tsx [hunk 2]
```diff
@@ -31,12 +31,12 @@ const InputField = ({ variant, text, required, ...rest }: InputProps) => {
...
     <div className='flex flex-col'>
       <label className='mt-3'> {required ? text + ' *' : text}</label>
-      <div className='flex justify-end content-center'>
+      <div className='flex content-center justify-end'>
         {variant === 'password' ? (
           <>
             <input className={variants} type={isPasswordShown ? 'text' : 'password'} {...rest} />
             <button
-              className='w-[22px] h-[15px] absolute mt-4 mr-3'
+              className='absolute mt-4 mr-3 h-[15px] w-[22px]'
```
Comment (joaquincollado) [inline src/components/shared/InputFields/index.tsx]: What did we say about unnecessary arbitrary values? 😡

File: src/pages/SignUp/signUp.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import SignUpForm from 'components/SignUpForm';
 
 const SignUp = () => {
   return (
-    <div className='bg-[url("/src/assets/background.jpg")] bg-cover h-[100vh] w-[100%] p-0 m-0 absolute'>
+    <div className='absolute m-0 h-[100vh] w-[100%] bg-[url("/src/assets/background.jpg")] bg-cover p-0'>
```
Comment (joaquincollado) [inline src/pages/SignUp/signUp.tsx]:
```suggestion
    <div className='absolute m-0 h-screen w-full bg-[url("/src/assets/background.jpg")] bg-cover p-0'>
```

**Review by joaquincollado — APPROVED** [review]
Great work Vale!

**Comment by diebarral** [comment]
Would be nice to see some screenshots or screen recordings

File: src/services/types.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export type UserType = {
```
Comment (diebarral) [inline src/services/types.ts]: Maybe you can create a `src/types` folder and create files for the different types

Comment (diebarral) [reply src/services/types.ts]: Also I have a few comments about this `UserType`:
* This looks more like an interface than a type, you can read more about this [here](https://blog.logrocket.com/types-vs-interfaces-in-typescript/)

* You could rewrite this like: 
```
type User = {
  pk: number;
  email: string;
  first_name: string;
  last_name: string;
}

interface UserState {
  access_token: string | null;
  refresh_token: string | null;
  user: UserType | null;
}
```
This way you can reuse the User type somewhere else

File: src/services/blackMarketApi.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { prepareHeaders } from 'utils/serviceUtils';
+
+export const blackMarketApi = createApi({
+  reducerPath: 'blackMarketAPI',
+  baseQuery: fetchBaseQuery({
+    baseUrl: process.env.REACT_APP_BASE_URL,
+    prepareHeaders,
+  }),
+  endpoints: (builder) => ({
+    postRegistration: builder.mutation({
```
Comment (diebarral) [inline src/services/blackMarketApi.ts]: I think `createUser` is a more suitable name for this mutation, what do you think? Same for the one below, just `logOut` makes more sense to me

File: src/App.css
Comment (DJA89) [inline src/App.css:1]: Weren't these needed for tailwind to work?

Comment (joaquincollado) [reply src/App.css:1]: The tailwind importation was duplicated in App.css and index.css

File: src/components/shared/InputFields/index.tsx (hunk 2 above)
Comment (ValesarroRS) [reply src/components/shared/InputFields/index.tsx]: 😿

**Pull request merged**